tcclient/creds.go

```diff
@@ -7,16 +7,75 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"net/http"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/taskcluster/jsonschema2go/text"
 	"github.com/taskcluster/slugid-go/slugid"
+	"github.com/tent/hawk-go"
 )
 
-type Credentials struct {
-	ClientId    string `json:"clientId"`
+type Credentials interface {
+	fmt.Stringer
+	ConfigureAuth(httpRequest *http.Request) error
+	getExtField() (string, error)
+	HawkCredentials() *hawk.Credentials
+}
+
+// Credentials for communicating with taskcluster services (Queue, Index,
+// Scheduler, etc).
+type TemporaryCredentials struct {
+	temporaryCredentials
+}
+
+type temporaryCredentials struct {
+	ClientID    string `json:"clientId"`
+	AccessToken string `json:"accessToken"`
+	// Certificate used only for temporary credentials
+	Certificate string `json:"certificate"`
+	// AuthorizedScopes if set to nil, is ignored. Otherwise, it should be a
+	// subset of the scopes that the ClientId already has, and restricts the
+	// Credentials to only having these scopes. This is useful when performing
+	// actions on behalf of a client which has more restricted scopes. Setting
+	// to nil is not the same as setting to an empty array. If AuthorizedScopes
+	// is set to an empty array rather than nil, this is equivalent to having
+	// no scopes at all.
+	// See http://docs.taskcluster.net/auth/authorized-scopes
+	AuthorizedScopes []string `json:"authorizedScopes"`
```

Comment:
I suspect the `reflect` package used by the `encoding/json` package will require the struct members to be exported in order to be able to marshal/unmarshal.